src/Standards/Squiz/Sniffs/PHP/NonExecutableCodeSniff.php

```diff
@@ -258,6 +258,12 @@ public function process(File $phpcsFile, $stackPtr)
                 continue;
             }
 
+            // Skip close PHP tag, followed by whitespace, followed by open PHP tag.
+            if ($tokens[$i]['code'] === T_INLINE_HTML && \trim($tokens[$i]['content']) === '') {
+                $i = $phpcsFile->findNext(T_OPEN_TAG, $i, $end);
+                continue;
+            }
```

Comment:
I believe this block of code will not work the way you think it does.

Please add the following code samples to the tests:
```php
<?php if (true): ?>
    <?php foreach ([] as $item): ?>
        <?php continue; ?>
        <div>non-executable</div>
    <?php endforeach; ?>
<?php endif; ?>

<?php if (true): ?>
    <?php foreach ([] as $item): ?>
        <?php continue; ?>

        <div>non-executable</div>

    <?php endforeach; ?>
<?php endif; ?>
```

The first one will still be reported (correctly), the second one will no longer report an issue, while it should.

Reply:
Yes, I can see this now. Thanks for highlighting the issue. I'll add these test cases and rework the suggestion to make it report correctly.

Reply:
I think I've fixed this now.